rpc/server.go

```diff
@@ -39,178 +40,84 @@ func NewRpcServer(nitroClient *nitro.Client, chainId *big.Int, logger zerolog.Lo
 
 	opts := &server.Options{}
 	ns, err := server.NewServer(opts)
-	handleError(err)
+	if err != nil {
+		panic(err)
+	}
 	ns.Start()
 
 	nc, err := nats.Connect(ns.ClientURL())
-	handleError(err)
+	if err != nil {
+		panic(err)
+	}
 
 	rs := &RpcServer{natstrans.NewNatsConnection(nc), ns, nitroClient, chainId, logger}
-	rs.registerHandlers()
+	err = rs.registerHandlers()
+	if err != nil {
+		panic(err)
+	}
+
 	return rs
 }
 
-// registerHandlers registers the handlers for the rpc server
-func (rs *RpcServer) registerHandlers() {
-	err := rs.connection.Subscribe(serde.DirectFundRequestMethod, func(data []byte) []byte {
-
+func subcribeToRequest[T serde.RequestPayload, U serde.ResponsePayload](rs *RpcServer, method serde.RequestMethod, processPayload func(T) U) error {
```

Comment:
Super Nit: would it be nice to place this function below `registerHandlers`? Might be a personal thing but I like to read the higher abstractions first and then drill down into the details.

Reply:
[8c18607](https://github.com/statechannels/go-nitro/pull/1117/commits/8c186077b892ee14ff8e08cc829fa843dcb7f229)